src/main/java/at/hannibal2/skyhanni/data/QuiverAPI.kt

```diff
@@ -0,0 +1,241 @@
+
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) {
+        val message = event.message.trimWhiteSpace().removeResets()
+
+        selectPattern.matchMatcher(message) {
+            val arrow = group("arrow")
+            currentArrow = getArrowByNameOrNull(arrow) ?: return
+            currentAmount = arrowAmount[currentArrow]?.toInt() ?: 0
+
+            return
+        }
+
+        fillUpJaxPattern.matchMatcher(message) {
+            val type = group("type")
+            val amount = group("amount").formatNumber().toFloat()
+
+            val filledUpType = getArrowByNameOrNull(type) ?: return
+
+            val existingAmount = arrowAmount[filledUpType] ?: 0f
+            val newAmount = existingAmount + amount
+            arrowAmount[filledUpType] = newAmount
+
+            return
+        }
+
+        fillUpPattern.matchMatcher(message) {
+            val flintAmount = group("flintAmount").formatNumber().toFloat()
+            val existingAmount = arrowAmount[getArrowByNameOrNull("ARROW".asInternalName())] ?: 0f
+            val newAmount = existingAmount + flintAmount
+
+            arrowAmount[getArrowByNameOrFlint("ARROW".asInternalName())] = newAmount
+
+            return
+        }
+
+        addedToQuiverPattern.matchMatcher(message) {
+            val type = group("type")
+            val amount = group("amount").formatNumber().toFloat()
+
+            val filledUpType = getArrowByNameOrNull(type) ?: return
+
+            val existingAmount = arrowAmount[filledUpType] ?: 0f
+            val newAmount = existingAmount + amount
+            arrowAmount[filledUpType] = newAmount
+
+            return
+        }
+
+        clearedPattern.matchMatcher(message) {
+            currentAmount = 0
+            arrowAmount.clear()
+
+            return
+        }
+
+        arrowResetPattern.matchMatcher(message) {
+            currentArrow = getArrowByNameOrNull("NONE".asInternalName()) ?: return
+            currentAmount = 0
+
+            return
+        }
+    }
+
+    @SubscribeEvent
+    fun onInventoryFullyLoaded(event: InventoryFullyOpenedEvent) {
+        if (!LorenzUtils.inSkyBlock) return
+        if (!quiverInventoryNamePattern.matches(event.inventoryName)) return
+
+        // clear to prevent duplicates
+        currentAmount = 0
+        arrowAmount.clear()
+
+        val stacks = event.inventoryItems
+        for (stack in stacks.values) {
+            val lore = stack.getLore()
+            if (lore.isEmpty()) continue
+
+            val arrow = stack.getInternalNameOrNull() ?: continue
+
+            val arrowType = getArrowByNameOrNull(arrow) ?: continue
+            val arrowAmount = stack.stackSize + this.arrowAmount.getOrDefault(arrowType, 0.0f)
+
+            this.arrowAmount[arrowType] = arrowAmount
+        }
+    }
+
+    @SubscribeEvent(priority = EventPriority.HIGHEST)
+    // Inspired by SkyblockFeatures - https://github.com/MrFast-js/SkyblockFeatures/
```

Comment:
If you have to use code from skyblock features (or other mods in general), please specify exactly which code you took and how you changed it.